src/app/prestador/prestador.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit } from '@angular/core';
+import { PrestadorService } from '../servicos/prestador.service';
+import { Router, ActivatedRoute } from '@angular/router';
+import { Prestador } from '../interfaces/prestador';
+import { ServicoDetalhado } from '../interfaces/servico-detalhado';
+import { Avaliacao } from '../interfaces/avaliacao';
+import { LocalStorageService } from '../servicos/local-storage.service';
+import { USER_TOKEN } from '../util/constantes';
+
+@Component({
+  selector: 'app-prestador',
+  templateUrl: './prestador.component.html',
+  styleUrls: ['./prestador.component.scss']
+})
+export class PrestadorComponent implements OnInit {
+  prestador:Prestador;
+  carregado:boolean = false;
+  constructor(private route:ActivatedRoute,
+    private router:Router,
+    private prestadorService:PrestadorService,
+    private localStorageService:LocalStorageService) { }
+
+  ngOnInit(): void {
+    this.route.queryParams.subscribe(params => {
+      let idPrestador:number = +params.idPrestador;
+      if(isNaN(idPrestador)) {
+        this.router.navigate(['/']);
+        return;
+      }
+      console.log(idPrestador)
+      this.localStorageService.getItem(USER_TOKEN).subscribe((token:string) => {
+        this.prestadorService.buscaPrestador(idPrestador, token).subscribe(prestador => {
+          this.carregado = true;
+          this.prestador = prestador;
+        });
+      })
+    })
+  }
+
+  getMediaAvaliacoes(servicoDetalhado:ServicoDetalhado) {
+    if(!servicoDetalhado.avaliacoes || servicoDetalhado.avaliacoes.length === 0) {
+      return 0;
+    }
+
+    let soma = servicoDetalhado.avaliacoes.reduce((soma:number, avaliacao:Avaliacao) => soma += avaliacao.media, 0);
```

Comment:
Codacy found an issue: [Arrow function should not return assignment.](https://app.codacy.com/manual/gaugsouza/petshow-frontend/pullRequest?prid=6332072)